### PR TITLE
Construction update

### PR DESCRIPTION
**Pull request by SinTwo**
Reworks how frames work to be more modular and adds the ability to construct and deconstruct:
- ATM
- Newscaster
- Guest pass console
- Status displays
- key card authenticator
- Fire Alarms
- Air Alarms
- Intercoms
- rechargers
- Photo copiers
- fax machines
- microwaves
- vendors (normal)
- large mining drills
- grinders
- Holopad
- conveyor belts
- wall rechargers
- entertainment monitors
- washing machines
- Notice boards
- mirrors
- IV
- roller beds
- shower curtains
- Fire extinguisher cabinets
- plastic flaps
- filling cabinets
- bookshelves
- water coolers

### CODE REVIEW

**Comment by Yoshax** [comment]
You left in your test box of parts with a comment saying it shouldn't be.

Also instead of moving things around into different branches you can use Squash to squash multiple commits into one.

So, this adds 3 new things you use to build with, gears, motors and springs. Which are producible in the autolathe, and spawn inside the engie vendor, yes?

**Comment by SinTwo** [comment]
Also I should have another comment next to the box comment saying I was leaving that in as you couldn't get the thing on-station anyway. I'll probably end up removing it after this has been in for a bit and settled.

Datraen and I attempted to squash the 20 some commits into 1 commit yesterday to no avail, it ended up being easier to just move the changed files into a new repository download and push that as a new branch.

The three new parts aren't needed for anything that's essential (air alarms, computers, etc.) They are producible through the autolathe for very low material costs.

File: code/game/machinery/autolathe_datums.dm [hunk 1]
```diff
@@ -171,6 +171,96 @@
...
+	name = "holopad electronics"
+	path = /obj/item/weapon/circuitboard/holopad
+	category = "Engineering"
+
+/datum/autolathe/recipe/guestpass
+	name = "guestpass console electronics"
+	path = /obj/item/weapon/circuitboard/guestpass
+	category = "Engineering"
+
+/*/datum/autolathe/recipe/entertainment
```
Comment (Datraen) [inline code/game/machinery/autolathe_datums.dm]: Still broken?

File: code/modules/paperwork/filingcabinet.dm [hunk 2]
```diff
@@ -46,6 +44,15 @@
 		playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
 		anchored = !anchored
 		user << "<span class='notice'>You [anchored ? "wrench" : "unwrench"] \the [src].</span>"
+	else if(istype(P, /obj/item/weapon/screwdriver))
+		if(do_after(user, 10))
```
Comment (Datraen) [inline code/modules/paperwork/filingcabinet.dm]: No informing the user that you're taking it apart beforehand?

File: code/game/machinery/frame.dm [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+								var/camt = min(M.amount, req_components[I]) // amount of cable to take, idealy amount required, but limited by amount provided
+								var/obj/item/stack/material/MS = new /obj/item/stack/material(src)
+								MS.amount = camt
+								MS.update_icon()
+								M.use(camt)
+								components += MS
+								req_components[I] -= camt
+								update_desc()
+								break
+								*/
```
Comment (Datraen) [inline code/game/machinery/frame.dm]: Why is this commented out? It's not needed, so it can be removed.

**Comment by Datraen** [comment]
:+1:

**Comment by Datraen** [comment]
:+1: on current implementation.

**Comment by Neerti** [comment]
No obvious faults I could find.

**Comment by Datraen** [comment]
And now the only thing we need is a wiki page for this.

**Comment by Yoshax** [comment]
Maybe it's just me, but I think the implantation of something big like this shouldn't be live until the relevant documentation for it exists.

**Pull request merged**